src/config/acbs.config.ts

```diff
@@ -0,0 +1,36 @@
+import { registerAs } from '@nestjs/config';
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+export interface AcbsConfig {
+  apiKey: string;
+  apiKeyHeaderName: string;
+
+  authentication: {
+    baseUrl: string;
+    clientId: string;
+    loginName: string;
+    password: string;
+  };
+
+  maxRedirects: number;
+  timeout: number;
+}
+
+export default registerAs(
+  'acbs',
+  (): AcbsConfig => ({
+    apiKey: process.env.ACBS_API_KEY,
+    apiKeyHeaderName: process.env.ACBS_API_KEY_HEADER_NAME,
+
+    authentication: {
+      baseUrl: process.env.ACBS_AUTHENTICATION_BASE_URL,
+      clientId: process.env.ACBS_AUTHENTICATION_CLIENT_ID,
+      loginName: process.env.ACBS_AUTHENTICATION_LOGIN_NAME,
+      password: process.env.ACBS_AUTHENTICATION_PASSWORD,
+    },
+
+    maxRedirects: parseInt(process.env.ACBS_MAX_REDIRECTS) || 21,
+    timeout: parseInt(process.env.ACBS_TIMEOUT) || 30000,
```

Comment:
The existing API has a timeout of 120 seconds which seems very long to me. I've set our default to 30 seconds but we can configure it as we need

Reply:
sounds good. we can change it if needed